### DISCUSSION TITLE
Problem with selfhosted Postgres + PGBounce

### DISCUSSION THREAD

**Discussion by Zwergenpunk** [plausible/analytics#1329] [Self-Hosted Support] [answered]
Hey, I ran into an issue wth an external postgres with pgbounce in front of it.

The website does run, but throws every 2nd or 3rd request a 500 error with the following error in the log:
```
** (Postgrex.Error) FEHLER 26000 (invalid_sql_statement_name) vorbereitete Anweisung »ecto_5642« existiert nicht
```

I think this is the error which is described here:
https://github.com/elixir-ecto/ecto_sql/issues/275

This could be fixable by: https://github.com/elixir-ecto/postgrex/#pgbouncer

I looked a short time into the code here, but I have no clue about elixir and I don't unterstand a word ;)

Is there some way to fix that?

**Comment by ukutaht** [comment, marked as answer]
I wonder what the downside of using unnamed prepared statements is. I'm thinking that if there were no downside to this, it would be the default setting in Postgrex.

We could certainly provide an option to use unnamed statements for Plausible users. We're always open for PRs, otherwise I have no idea when I'll get to adding it :/

**Reply by Zwergenpunk** [reply]
Yeah, there is probably a downside. But if every second request fails there is also a downside ;)
We try to fix it with adjusting pgbouncer.

If that does not work, I will try to make a PR and add a new database env variable

**Reply by ukutaht** [reply]
Cheers, thanks!

**Reply by bartoszhernas** [reply]
@Zwergenpunk have you fixed this somehow?

**Reply by Zwergenpunk** [reply]
> @Zwergenpunk have you fixed this somehow?

Hey, no we adjusted pgbouncer to allow those statements.

**Reply by bartoszhernas** [reply]
Ok, thanks :) I guess you switched pgbouncer to "session" mode? I need to keep it in transaction mode, so plausible for now will connect directly. Metabase uses JDBC so I fixed it by adding "prepareThreshold=0" to connection  string
